core/lib/tasks/migrations/migrate_shipping_rate_taxes.rake

```diff
@@ -0,0 +1,17 @@
+namespace :solidus do
+  namespace :migrations do
+    namespace :migrate_shipping_rate_taxes do
+      task up: :environment do
+        puts "Adding persisted tax notes to historic shipping rates"
+        Spree::ShippingRate.where.not(tax_rate_id: nil).find_each do |shipping_rate|
+          tax_rate = Spree::TaxRate.unscoped.find_by(id: shipping_rate.tax_rate_id)
+          shipping_rate.taxes.find_or_create_by!(
+            tax_rate: tax_rate,
+            amount: tax_rate.compute_amount(shipping_rate)
+          )
+        end
+        Spree::ShippingRate.where.not(tax_rate_id: nil).update_all(tax_rate: nil)
```

Comment:
This seems odd to have inside the loop. Should this just be updating `shipping_rate` or should this be outside the loop?